server/routes/approvingLicences/handlers/approvalCases.ts

```diff
@@ -2,16 +2,23 @@ import { Request, Response } from 'express'
 import moment from 'moment'
 import _ from 'lodash'
 import CaseloadService from '../../../services/caseloadService'
+import PrisonerService from '../../../services/prisonerService'
 import { convertToTitleCase } from '../../../utils/utils'
 
 export default class ApprovalCaseRoutes {
-  constructor(private readonly caseloadService: CaseloadService) {}
+  constructor(private readonly caseloadService: CaseloadService, private readonly prisonerService: PrisonerService) {}
 
   GET = async (req: Request, res: Response): Promise<void> => {
     const search = req.query.search as string
 
     const { user } = res.locals
-    const cases = await this.caseloadService.getApproverCaseload(user)
+    const { caseloadsSelected = [] } = req.session
+    const hasMultipleCaseloadsInNomis = user.prisonCaseload.length > 1
+    const allPrisons = await this.prisonerService.getPrisons()
+    const activeCaseload = allPrisons.filter(p => p.agencyId === user.activeCaseload)
+    const prisonCaseloadToDisplay = caseloadsSelected.length ? caseloadsSelected : [activeCaseload[0].agencyId]
```

Comment:
This is a lot of stuff to be doing in a controller and presumably you have similar stuff in the CA flow.

do you have a userservice you could push a lot of this down to?  getSelectedPrisons or something?

Reply:
Actually it might be good to enrich the user and their caseloads with prison info in the `populateCurrentUser` middleware.
It already called prison api and it only does it once per session rather than on every request.